services/shuttle-crontab/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use axum::Router;
+use chrono::Utc;
+use cron::Schedule;
+use router::make_router;
+use serde::{Deserialize, Serialize};
+use shuttle_persist::PersistInstance;
+use shuttle_runtime::tracing::{debug, info};
+use tokio::{
+    sync::mpsc::{self, Receiver, Sender},
+    sync::oneshot,
+    time::sleep,
+};
+
+mod error;
+use error::CrontabServiceError;
+
+mod router;
+
+pub type ShuttleCrontab = Result<CrontabService, shuttle_runtime::Error>;
+
+type Responder<T> = oneshot::Sender<Result<T, CrontabServiceError>>;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RawJob {
+    schedule: String,
+    url: String,
+}
+
+#[derive(Debug)]
+pub enum Msg {
+    NewJob(RawJob, Responder<()>),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Crontab {
+    jobs: Vec<RawJob>,
+}
+
+#[derive(Debug)]
+pub struct CronJob {
+    schedule: Schedule,
+    url: String,
+}
+
+impl CronJob {
+    fn from_raw(raw: &RawJob) -> Self {
+        let schedule = Schedule::from_str(&raw.schedule).expect("Failed to parse schedule");
+        Self {
+            schedule,
+            url: raw.url.clone(),
+        }
+    }
+
+    async fn run(&self) {
+        debug!("Running job for: {}", self.url);
+        while let Some(next_run) = self.schedule.upcoming(Utc).next() {
+            let next_run_in = next_run
+                .signed_duration_since(chrono::offset::Utc::now())
+                .to_std()
+                .unwrap();
+            sleep(next_run_in).await;
+
+            info!("Calling {}", self.url);
+
+            let req = reqwest::get(self.url.clone()).await.unwrap();
+            info!("Called with status code {}", req.status());
+        }
+    }
+}
+
+pub struct CronRunner {
+    persist: PersistInstance,
+    receiver: Receiver<Msg>,
+}
+
+impl CronRunner {
+    async fn run_jobs(&mut self) {
+        if let Ok(tab) = self.persist.load::<Crontab>("crontab") {
+            debug!("Found {} jobs", tab.jobs.len());
+            for raw in tab.jobs {
+                debug!("Starting job: {:?}", raw);
+                let job = CronJob::from_raw(&raw);
+
+                tokio::spawn(async move {
+                    job.run().await;
+                });
```

Comment:
This doesn't seem to work any other way, since we need the `move` to not run into lifetime issues (We do want to completely move this to another thread after all).